app/Http/Controllers/QuizAPIController.php

```diff
@@ -51,4 +49,24 @@ function addQuiz(Request $request)
             ], 500);
         }
     }
+
+    function singleQuiz(int $id)
+    {
+        try {
+            $quiz = Quiz::with(['questions' => ['answers'],])->find($id);
+        } catch (Exception $e) {
+            $quiz = null;
```

Comment:
If you catch here, you're having an internal server error. Is 404 the code for internal server error?